libcudacxx/include/cuda/std/__concepts/concept_macros.h

```diff
@@ -0,0 +1,278 @@
+//===----------------------------------------------------------------------===//
+//
+// Copyright (c) Facebook, Inc. and its affiliates.
+// Copyright (c) 2022 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Part of libcu++, the C++ Standard Library for your entire system,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2023 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _CUDA___CONCEPTS
+#define _CUDA___CONCEPTS
+
+#include <cuda/std/detail/__config>
+
+#if defined(_CCCL_IMPLICIT_SYSTEM_HEADER_GCC)
+#  pragma GCC system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_CLANG)
+#  pragma clang system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_MSVC)
+#  pragma system_header
+#endif // no system header
+
+////////////////////////////////////////////////////////////////////////////////
+// _CCCL_TEMPLATE
+// Usage:
+//   _CCCL_TEMPLATE(typename A, typename _Bp)
+//     _CCCL_REQUIRES( Concept1<A> _CCCL_AND Concept2<_Bp>)
+//   void foo(A a, _Bp b)
+//   {}
+
+// Barebones enable if implementation to use outside of cuda::std
+template <bool>
+struct __cccl_select
+{};
+
+template <>
+struct __cccl_select<true>
+{
+  template <class _Tp>
+  using type = _Tp;
+};
+
+template <bool _Bp, class _Tp = void>
+using __cccl_enable_if_t = typename __cccl_select<_Bp>::template type<_Tp>;
```

Comment:
Q: We are not using `enable_if` here, because this header is way higher up in the include chain and we don't want to depend on `<type_traits>` here?